lib/kafka_ex/protocol/sync_group.ex

```diff
@@ -0,0 +1,76 @@
+defmodule KafkaEx.Protocol.SyncGroup do
+  import KafkaEx.Protocol.Common
+
+  @member_assignment_version 0
+
+  defmodule Assignment do
+    defstruct topic: nil, partitions: []
+    @type t :: %Assignment{topic: binary, partitions: [integer]}
+  end
+
+  defmodule Response do
+    defstruct error_code: nil, assignments: []
+    @type t :: %Response{error_code: atom | integer, assignments: [Assignment.t] }
+  end
+
+  @spec create_request(integer, binary, binary, integer, binary, [{binary, [Assignment.t]}]) :: binary
+  def create_request(correlation_id, client_id, group_name, generation_id, member_id, assignments) do
+    KafkaEx.Protocol.create_request(:sync_group, correlation_id, client_id) <>
+    << byte_size(group_name) :: 16-signed, group_name :: binary,
+       generation_id :: 32-signed,
+       byte_size(member_id) :: 16-signed, member_id :: binary,
+       length(assignments) :: 32-signed, group_assignment_data(assignments, "") :: binary,
+       0 :: 32-signed
+    >>
+  end
+
+  @spec parse_response(binary) :: Response.t
+  def parse_response(<< _correlation_id :: 32-signed, error_code :: 16-signed,
+                        _member_assignment_len :: 32-signed,
```

Comment:
This would mean we need extra indirections in the parsing here, further complicating the parsing code. As long as we don't have error-handling for short packets anywhere else, I'm not sure that this complication warrants the work.